crates/pera-core/src/consensus_handler.rs

```diff
@@ -355,6 +355,63 @@ impl<C: CheckpointServiceNotify + Send + Sync> ConsensusHandler<C> {
                             .stats
                             .inc_num_user_transactions(authority_index as usize);
                     }
+
+                    // If we receive a `DwalletMPCOutput` transaction,
+                    // verify that it's valid and create a system transaction
+                    // to store its output on the blockchain,
+                    // so it will be available for the initiating user.
+                    if let ConsensusTransactionKind::DwalletMPCOutput(
+                        value,
+                        session_id,
+                        initiating_address,
+                    ) = &transaction.kind
+                    {
+                        info!(
+                            "Received proof mpc output from authority {:?} for session {:?}",
+                            authority_index, session_id
+                        );
+
+                        let mpc_manager = self.epoch_store.proof_mpc_manager.get();
+                        let is_valid_transaction = match mpc_manager {
+                            Some(mpc_manager) => {
+                                mpc_manager.lock().await
+                                    .try_verify_output(value, session_id)
+                                    .unwrap_or_else(|e| {
+                                        // TODO (#311): Make validator don't mark other validators as malicious or take any active action while syncing
+                                        error!("error verifying ProofMPCOutput for the session {:?} from party {:?}: {:?}",session_id, authority_index, e);
+                                        false
+                                    })
```

Comment:
In line 411, referring to TODO (#263), you do have a TODO to handle malicious behaviour, can you clarify this comment? Are you saying here you just log for debugging and the malicious behavious is handled later on? I'd rather say it like this, explain why, and refer to the point where this is done (line 411 probably).

Reply:
The TODOs are here to avoid repeated questions.
You can go to issues like this: https://github.com/dwallet-labs/dwallet-network/issues/263
and see the comments there, this will lead you to the following comment:

<img width="809" alt="image" src="https://github.com/user-attachments/assets/00c0e485-0a90-46f7-9499-49a461f33c35">